### PR TITLE
[prometheus] Fix AlertManager deploy.yaml missing `cluster.listen-address`

### PR DESCRIPTION
**Pull request by vigohe**
#### What this PR does / why we need it:

* Fix in prometheus chart `alertmanager/deploy.yaml` missing `cluster.listen-address` when `alertmanager.service.enableMeshPeer` it's `true`.

Alertmanager `--cluster.listen-address` default value is `0.0.0.0:9094` so we need to overwrite it to `0.0.0.0:6783`

* Fix in prometheus chart when adding multiple `clusterPeer=` at Alertmanager `extraArgs`

example:

```
alertmanager.extraArgs:
  cluster.peer: "foo:6783"
  cluster.peer: "bar:6783"
```

Because extraArgs it's defined as a Map, Helm will only render the last `clusterPeer` defined when executing the `range` command

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name

### CODE REVIEW

**Review by zanhsieh — APPROVED** [review]

File: charts/prometheus/templates/alertmanager/deploy.yaml [hunk 1]
```diff
@@ -60,7 +60,12 @@ spec:
           args:
             - --config.file=/etc/config/{{ .Values.alertmanager.configFileName }}
             - --storage.path={{ .Values.alertmanager.persistentVolume.mountPath }}
+          {{- if .Values.alertmanager.service.enableMeshPeer }}
+            - --cluster.listen-address=0.0.0.0:6783
             - --cluster.advertise-address=[$(POD_IP)]:6783
+          {{- else }}
+            - --cluster.listen-address=
```
Comment (monotek) [inline charts/prometheus/templates/alertmanager/deploy.yaml:67]: Should this be empty?

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus/templates/alertmanager/deploy.yaml (hunk 1 above)
Comment (vigohe) [reply charts/prometheus/templates/alertmanager/deploy.yaml:67]: It’s to turn off HA on Alertmanager

ref: https://github.com/prometheus/alertmanager#turn-off-high-availability

**Review by ordenador — APPROVED** [review]

**Review by abrahamjoc — APPROVED** [review]

**Comment by vigohe** [comment]
@zanhsieh I found another Issue when you define multiple `clusterPeer` at `alertmanager.extraArgs`, I added the fix in this PR

File: charts/prometheus/templates/alertmanager/deploy.yaml (hunk 1 above)
Comment (vigohe) [reply charts/prometheus/templates/alertmanager/deploy.yaml:67]: @monotek  any comments ?

**Comment by vigohe** [comment]
@zanhsieh @monotek can you review it please,  we have `alertmanagers` installed in more than one region so we need this fix.

Thanks!

**Review by ordenador — DISMISSED** [review]
LGTM

**Review by zanhsieh — DISMISSED** [review]

**Review by zanhsieh — APPROVED** [review]

**Review by ordenador — DISMISSED** [review]

**Comment by torstenwalter** [comment]
@monotek could you re-review?

**Review by monotek — APPROVED** [review]

**Pull request merged**